resources/todomvc/big-dom-generator/src/components/tree-area.jsx

```diff
@@ -3,26 +3,30 @@ import { useRef } from "react";
 import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_NUMBER_OF_CHILDREN, PROBABILITY_OF_HAVING_CHILDREN, TARGET_SIZE, MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH } from "../../params";
 
 import ChevronRight from "./../assets/Smock_ChevronRight_18_N.svg";
-import TaskListIcon from "./../assets/Smock_TaskList_18_N.svg";
 
 const FolderWrapper = (props) => {
     const { nodeCount, random, maxDepth, maxNumChildren, childProbability, currentDepth } = props;
+
     // prettier-ignore
-    if (currentDepth >= maxDepth)
+    if (nodeCount.current >= TARGET_SIZE || currentDepth >= maxDepth)
         return null;
+
+    nodeCount.current = nodeCount.current + 1;
+
     // Choose a random number of children.
     const numChildren = random.randRange(1, maxNumChildren);
     const children = [];
-    for (let i = 0; i < numChildren && nodeCount.current < TARGET_SIZE; i++)
+    for (let i = 0; i < numChildren && nodeCount.current < TARGET_SIZE; i++) {
         children.push(<TreeItem key={i} nodeCount={nodeCount} random={random} numChildren={numChildren} maxNumChildren={maxNumChildren} maxDepth={maxDepth} childProbability={childProbability} currentDepth={currentDepth + 1} />);
+        nodeCount.current = nodeCount.current + 9;
```

Comment:
Will this work as expected? IIUC these children could be null if we happened to reach the target size before they are rendered. Is there something wrong with calling this inside the `TreeItem` component?

Reply:
### Using the code on the left side before my changes in this scenario.

Suppose `MAX_NUMBER_OF_CHILDREN=16`, `TARGET_SIZE=10` and `random.randRange(1, maxNumChildren);` always returns **14**.

The `<FolderWrapper />` element will then get a prop of a `maxNumberOfChildren={16}`.

In line 14, `numChildren` gets a value of 14 from `random.randRange(1, maxNumChildren);`. 

We will push in 14 `<TreeItem />` and already violate the constraint before rendering the other elements inside TreeItem by the `<li>` elements alone.

Reply:
There are some elements that Treeitem will always render regardless of if we reached the limit or not.

Reply:
What if we query for nodeCount.current at the beginning of TreeITem and return null if it already reached the target size

Reply:
Just pushed some code, with TARGET_SIZE=100 it will return 103 elements in tree-area.

TARGET_SIZE=200 will return 204. seems good to me